packages/cra-template/template/src/App.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { Stack, Text, Link, FontWeights, IStackTokens } from '@fluentui/react';
+import logo from './logo.svg';
+import './App.css';
+
+const boldStyle = { root: { fontWeight: FontWeights.semibold } };
+const stackTokens: IStackTokens = { childrenGap: 15 };
+
+export const App: React.FunctionComponent = () => {
+  return (
+    <Stack
+      horizontalAlign="center"
+      verticalAlign="center"
+      verticalFill
+      styles={{
```

Comment:
> creating styling objects inline causes perf issues

That might be questionable :) JIT is quite powerful and having to evaluate same static object multiple times (hot path tends to be quite cheap). Also React won't re-render physical DOM as nothing changed in diff tree :) 

I'd suggest follow the common approach which is - measure first, optimise if needed

Reply:
@Hotell I think @ecraig12345 meant these guidelines: https://github.com/microsoft/fluentui/wiki/Performance

Reply:
thanks for link @kubkon 💪. As I mentioned it's not mutated on every render. 

Btw don't we have lint rules for this ? From my experience if it's not caught by tooling it's hard to "encourage/enforce" so user follows best practices. 

anyways great job !

Reply:
While I agree that measure first/optimize if needed is a good approach, this is a case where we've already measured and definitely know it to be a problem. The literal object creation isn't the problem, it's the caching changes it causes. Note that the prop here is not built-in `style`, it's `styles`, which is a custom prop that runs through merge-styles. On every re-render of this component, a new object will be created, causing the merge-styles cache to be invalidated and all calculations to be re-run. In real apps this is a well known cause of perf problems (the guidelines @kubkon linked).

Agreed that this would be a good place to have tooling, like a custom lint rule. But until someone implements that, PR review comments are the next best thing.